### ISSUE TITLE
mraplot is exported and thus required to exist when using MADNESS externally

### ISSUE THREAD

**Issue by mbanck** [m-a-d-n-e-s-s/madness#226]
Contrary to the other four binaries installed into /usr/bin (```tdse, tdse4, moldft, mcpfit```), ```mraplot``` is also exported, see ```src/madness/mra/CMakeLists.txt```:

```
add_executable(mraplot mraplot.cc)
target_link_libraries(mraplot MADmra)

# Install the MADmra library
install(TARGETS mraplot EXPORT madness
    RUNTIME DESTINATION "${MADNESS_INSTALL_BINDIR}"
    COMPONENT mra)
```

Also note the comment "Install the MADmra library", is this maybe a typo? If not, what is the rationale to include mraplot in madness-targets.cmake as it seems to be a utility program? This means it is required to be present if one wants to develop against MADNESS, which looks odd to me. If I try to build TiledArray without mraplot present, I get:

```
CMake Error at /usr/lib/x86_64-linux-gnu/cmake/madness/madness-targets.cmake:150 (message):
  The imported target "mraplot" references the file

     "/usr/bin/mraplot"

  but this file does not exist.  Possible reasons include:
```

**Comment by robertjharrison** [comment]
Thanks for pointing this out and your other comments.  We are poised to do
a major source reorganization and to move a bunch of the peripheral tools
and applications into a separate distribution so that we can focus on
having a consistent and higher quality core.

Can I inquire your interest in madness?   We are always happy to have new
contributors and users, and you seem much better informed than most!

Best wishes

Robert

**Comment by mbanck** [comment]
I'm packaging it for Debian, mainly as it is a dependency of TiledArray which in turn enhances MPQC3, which we've packaged in Debian for a long time.

I'm using this patch now, though I am not very clear on whether the ```COMPONENT mra``` is correct/needed.

```
--- ./src/madness/mra/CMakeLists.txt.orig       2016-12-29 14:28:07.352387512 +0100
+++ ./src/madness/mra/CMakeLists.txt    2016-12-29 14:30:14.676171248 +0100
@@ -19,11 +19,7 @@
 # Create executables
 add_executable(mraplot mraplot.cc)
 target_link_libraries(mraplot MADmra)
-
-# Install the MADmra library
-install(TARGETS mraplot EXPORT madness
-    RUNTIME DESTINATION "${MADNESS_INSTALL_BINDIR}"
-    COMPONENT mra)
+install(TARGETS mraplot DESTINATION "${MADNESS_INSTALL_BINDIR}" COMPONENT mra)
 install(FILES autocorr coeffs gaussleg
     DESTINATION "${MADNESS_INSTALL_DATADIR}"
     COMPONENT mra)
```

**Comment by evaleev** [comment]
@mbanck yes, mraplot should not be exported, only installed. This is a mistake on our part, this was probably a carryover from the silly autotools harness where such differences do not matter.

MPQC4/TiledArray are the only users of installed MADNESS and they only use the runtime + some infrastructure bits (e.g. BLAS/LAPACK bindings, out of sheer laziness on our part). We do not use mra yet although we are exploring this as a part of the MPQC4 design.

**Comment by evaleev** [comment]
resolved by d5a7ce75fe3ab8794dbabaf6c86d4709fce22b5d
